test/rulesets/LinearAlgebra/norm.jl

```diff
@@ -39,6 +39,26 @@
             @test extern(rrule(fnorm, zero(x))[2](ȳ)[2]) ≈ zero(x)
             @test rrule(fnorm, x)[2](Zero())[2] isa Zero
         end
+        ndims(x) > 1 && @testset "non-strided" begin
+            xp = if x isa Matrix
+                view(x, [1,2,3], 1:3)
+            elseif x isa Array{T,3}
+                PermutedDimsArray(x, (1,2,3))
+            end
+            @test !(xp isa StridedArray)
+            y = fnorm(x)
+            # ẋ = rand(T, size(xp)) # rand_tangent(xp)
+            x̄ = rand(T, size(xp)) # rand_tangent(xp)
+            ȳ = rand_tangent(y)
+            # frule_test(fnorm, (xp, ẋ))
+            rrule_test(fnorm, ȳ, (xp, x̄))
+        end
+        T == Float64 && ndims(x) == 1 && @testset "Integer input" begin
+            x = [1,2,3]
+            _, int_back = rrule(fnorm, x)
+            _, float_back = rrule(fnorm, float(x))
+            @test unthunk(int_back(1.0)[2]) ≈ unthunk(float_back(1.0)[2])
```

Comment:
Can we test this for inplace accumulation as well?

Reply:
The particular bug I wanted to protect against is allocating `dx = similar(x)`, which is too narrow a type. For the in-place thunk, I actually have no idea where the machinery to allocate `dx` lives & how it decided the type, is that part of this package or the AD's job? (I'm not sure if anyone uses these yet, either!)

Reply:
Ah, I see. Since it's just testing that case, I am fine with not testing inplace accumulation. `dx` will probably end up being based on the output shape of the forward pass, so it's not impossible there might be some issues with `Inf` and `NaN` and `Diagonal` outputs, but it's probably not worth worrying about too much for now. If you were to ask me, I am not sure I'm convinced having `NaN * ::Diagonal` return a dense matrix is really worth it, since we have the notion of "strong zeroes" in other places in LinearAlgebra as well, but it is what it is.

Reply:
OK. IMO we should treat `NaN * ::Diagonal` as a bug in LinearAlgebra to be fixed, but it won't be tomorrow.